iaso/api/permissions.py

```diff
@@ -38,9 +38,13 @@ def grouped_permissions(self, request):
         perms = self.queryset(request)
         result = {}
         for group in PERMISSIONS_PRESENTATION.keys():
-            result[group] = []
-            for permission in perms.filter(codename__in=PERMISSIONS_PRESENTATION[group]):
-                result[group].append({"id": permission.id, "name": _(permission.name), "codename": permission.codename})
+            permissions = perms.filter(codename__in=PERMISSIONS_PRESENTATION[group])
+            if permissions:
+                result[group] = []
+                for permission in permissions:
+                    result[group].append(
+                        {"id": permission.id, "name": _(permission.name), "codename": permission.codename}
+                    )
```

Comment:
```suggestion
            for permission in perms.filter(codename__in=PERMISSIONS_PRESENTATION[group]):
                result[group] = []
                result[group].append({"id": permission.id, "name": _(permission.name), "codename": permission.codename})
```

I think you can keep the previous code and just move the `result[group] = []` line inside the `for` loop.

This could save a bunch of statements.

Reply:
But at each loop it will remove the previous added permission and at the end result[group] will have the last permission

Reply:
Yes you're right!
But that's strange because I ran the unit tests with this modified code and they were successful 🤔

Reply:
`self.assertTrue(
            set([permission["codename"] for permission in response.json()["permissions"]["org_units"]])
            <= set(PERMISSIONS_PRESENTATION["org_units"]),
        )`. Maybe I did it wrong here?